### ISSUE TITLE
Analysis or Edge support_graphs?

### ISSUE THREAD

**Issue by edeutsch** [NCATSTranslator/ReasonerAPI#434]
In our TRAPI 1.4 Migration Guide, it shows the use of Edge support_graphs (via an attribute):
```
"creative_edge": {
    "subject": "metformin",
    "object": "diabetes",
    "predicate": "treats",
    "attributes": [
        {
            "attribute_type_id": "biolink:support_graphs",
            "values": [
                "a0"
            ]
        }
    ]
}
```

as well as Analysis support_graphs via a property:
```
{
    "resource_id": "infores:ara0"
    "edge_bindings": {
        "e0": [
            {
                "id": "e01"
            }
        ]
    }
    "support_graphs": [
        "a1",
        "a2"
    ],
    "score": .6
}
```

For our one-hop creative mode queries that we are currently working with (with essentially one Edge in each Result), it appears that some agents are using edge support_graphs and some are doing Analysis support_graphs. This might cause some difficulties down the road.

I am wondering if we should articulate some guidance on when to use Edge support_graphs and when to use Analysis support_graphs, especially for 1-hop creative queries, which is our bread and butter for now.

is there already guidance?

Should we create some?

Or just let agents do whatever?

**Comment by kaiwenho** [comment]
Unsecret is only using the Edge support_graphs, since we are generating inferred/creative edges by expanding a chain starting from the subject and ending at the object, which is similar to the "creative_edge" Edge in the example. I am feeling that the choice of support_graphs is based on the need. In the example, "creative_edge" is generated by the ARA while "e01" should come from a concrete KG/KP/source. When the "creative_edge" is generated, it is necessary to immediately point out the 'rationale' (through Edge support_graphs), hence using Edge support_graphs is straightforward with this kind of edge. However, the Analysis support_graphs is more like supporting a result instead of an edge. In the example, both "creative_edge" and "e01" are merged into one analysis object. Hence "a1" and "a2" should be supporting the result where there are two edges included. Therefore, using which support_graphs should depend on which object (Edge/Result) the auxiliary graph is supporting. 

The example I was referring to is https://github.com/uhbrar/ReasonerAPI/blob/update_guide/MigrationAndImplementationGuide1-4.md#trapi-14-example-message

This is my interpretation of the example message.

**Comment by uhbrar** [comment]
Both Kaiwen and Chris are mostly in line with my thinking. Support graphs on edges are evidence for the edge, and as such. Support graphs in an analysis block are used in the calculation of a score, and thus inform the score.

I agree that the documentation should probably be a bit more clear on that.

**Comment by edeutsch** [comment]
Addressed in #435. closing
